pkg/apis/numaplane/v1alpha1/shared_types.go

```diff
@@ -73,3 +74,15 @@ type ProgressiveStrategy struct {
 	// if ForcePromote is set, assessment will be skipped and Progressive upgrade will succeed
 	ForcePromote bool `json:"forcePromote,omitempty"`
 }
+
+// Rider defines a resource that can be deployed along with the primary child of a rollout
+type Rider struct {
+	// if set, perform progressive rollout when it changes
+	Progressive bool `json:"progressive,omitempty"`
+
+	// if set, rider resource should be created for each vertex
+	PerVertex bool `json:"perVertex,omitempty"`
```

Comment:
this field doesn't make sense for isbsvc and monovertex

Can you separate this into 2 separate structs:
1. `Rider` is what you have here except remove `PerVertex` field
2. `PipelineRider`:
```
type PipelineRider struct {
    Rider `json:",inline"`
    // if set, rider resource should be created for each vertex
    PerVertex bool `json:"perVertex,omitempty"`
}
```